pkg/controller/rolloutblock/rolloutblock_controller.go

```diff
@@ -0,0 +1,622 @@
+	}
+
+	key, err := cache.MetaNamespaceKeyFunc(rb)
+	if err != nil {
+		runtime.HandleError(err)
+		return
+	}
+
+	rolloutBlockUpdater := RolloutBlockUpdater{
+		key,
+		NewRolloutBlockObject,
+		"",
+		true,
+	}
+
+	c.rolloutblockWorkqueue.Add(rolloutBlockUpdater)
+}
+
+// Run starts RolloutBlock controller workers and blocks until stopCh is
+// closed.
+func (c *Controller) Run(threadiness int, stopCh <-chan struct{}) {
+	defer runtime.HandleCrash()
+	defer c.rolloutblockWorkqueue.ShutDown()
+
+	glog.V(2).Info("Starting RolloutBlock controller")
+	defer glog.V(2).Info("Shutting down RolloutBlock controller")
+
+	if ok := cache.WaitForCacheSync(
+		stopCh,
+		c.applicationsSynced,
+		c.releasesSynced,
+		c.rolloutBlockSynced,
+	); !ok {
+		runtime.HandleError(fmt.Errorf("failed to wait for caches to sync"))
+		return
+	}
+
+	for i := 0; i < threadiness; i++ {
+		go wait.Until(c.runRolloutBlockWorker, time.Second, stopCh)
+	}
+
+	glog.V(4).Info("Started RolloutBlock controller")
+
+	<-stopCh
+}
+
+func (c *Controller) runRolloutBlockWorker() {
+	for c.processNextRolloutBlockWorkItem() {
+	}
+}
+
+func (c *Controller) processNextRolloutBlockWorkItem() bool {
+	obj, shutdown := c.rolloutblockWorkqueue.Get()
+	if shutdown {
+		return false
+	}
+
+	defer c.rolloutblockWorkqueue.Done(obj)
+
+	var (
+		ok                  bool
+		rolloutBlockUpdater RolloutBlockUpdater
+	)
+
+	if rolloutBlockUpdater, ok = obj.(RolloutBlockUpdater); !ok {
+		c.rolloutblockWorkqueue.Forget(obj)
+		runtime.HandleError(fmt.Errorf("invalid object key (will retry: false): %#v", obj))
+		return true
+	}
+
+	shouldRetry := false
+	err := c.syncRolloutBlock(rolloutBlockUpdater)
+
+	if err != nil {
+		shouldRetry = shippererrors.ShouldRetry(err)
+		runtime.HandleError(fmt.Errorf("error syncing RolloutBlock %q (will retry: %t): %s", rolloutBlockUpdater.RolloutBlockKey, shouldRetry, err.Error()))
+	}
+
+	if shouldRetry {
+		if c.rolloutblockWorkqueue.NumRequeues(rolloutBlockUpdater) >= maxRetries {
+			// Drop this update out of the workqueue and thus reset its
+			// backoff. This limits the time a "broken" object can hog a worker.
+			glog.Warningf("Update %q for RolloutBlock has been retried too many times, dropping from the queue", rolloutBlockUpdater)
+			c.rolloutblockWorkqueue.Forget(rolloutBlockUpdater)
+
+			return true
+		}
+
+		c.rolloutblockWorkqueue.AddRateLimited(rolloutBlockUpdater)
+
+		return true
+	}
+
+	glog.V(4).Infof("Successfully synced RolloutBlock Updater %q", rolloutBlockUpdater)
+	c.rolloutblockWorkqueue.Forget(obj)
+
+	return true
+}
+
+func (c *Controller) syncRolloutBlock(rolloutBlockUpdater RolloutBlockUpdater) error {
+	key := rolloutBlockUpdater.RolloutBlockKey
+	updaterType := rolloutBlockUpdater.UpdaterType
+	if updaterType == NewRolloutBlockObject {
+		return c.syncNewRolloutBlockObject(key)
+	}
+
+	overridingKey := rolloutBlockUpdater.OverridingObjectKey
+	if len(key) == 0 || len(overridingKey) == 0 {
+		return nil
+	}
+
+	var err error = nil
```

Comment:
No need for an explicit nil assignment:
```go
var err error
```
As it's going to be set to nil by default.